addons/mrp/models/mrp_production.py

```diff
@@ -200,6 +200,13 @@ def _get_default_location_dest_id(self):
     production_location_id = fields.Many2one('stock.location', "Production Location", related='product_id.property_stock_production', readonly=False)
     picking_ids = fields.Many2many('stock.picking', compute='_compute_picking_ids', string='Picking associated to this manufacturing order')
     delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_ids')
+    is_ready_produce = fields.Boolean(compute='_compute_is_ready_produce')
+
+    @api.depends('move_raw_ids', 'move_raw_ids.reserved_availability')
+    def _compute_is_ready_produce(self):
+        for order in self:
+            if order.state not in ('draft', 'done', 'cancel'):
+                order.is_ready_produce = all(move.bom_line_id.product_qty <= move.reserved_availability for move in order.move_raw_ids)
```

Comment:
You just need to check the state of move with a bilt-in methode: https://github.com/odoo/odoo/blob/master/addons/stock/models/stock_move.py#L588

```python
for order in self.filtered(lambda x: x.state not in ('draft', 'done', 'cancel')):
     order.has_ready_produce = order.move_raw_ids._get_relevant_state_among_moves() == 'assigned'
```

Reply:
@jip-odoo, what do you think ?

Reply:
We don't need moves with full availability if partial available then also we can move to production.

Reply:
use also `partially_available`
` order.move_raw_ids._get_relevant_state_among_moves() in ('assigned', 'partially_available')`

Reply:
@fmdl @sle-odoo Can we start production without minimum quantity requirement satisfied ?
if yes then we will go with state...

Reply:
it depends on a settings on the bom right? if the move is partially available but has enough reserved components to produce 1 unit, then it should be ready